src/core/cuda_memory_manager.cc

```diff
@@ -57,6 +57,7 @@ PointerToString(void* ptr)
 namespace nvidia { namespace inferenceserver {
 
 std::unique_ptr<CudaMemoryManager> CudaMemoryManager::instance_;
+std::mutex CudaMemoryManager::cuda_memory_creating_;
```

Comment:
Call this 'instance_mu_' or similar because I think it should also be used in the Reset call to protect intantance_.reset()